lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -961,6 +964,27 @@ def set_http_request(record, entry)
       output.cache_validated_with_origin_server = \
         cache_validated_with_origin_server \
         unless cache_validated_with_origin_server.nil?
+
+      latency = input.delete('latency')
+      unless latency.nil?
+        # Parse latency to duration. If failed, skip setting latency.
+        duration_in_seconds = ChronicDuration.parse(latency)
+        if duration_in_seconds
+          if @use_grpc
+            # Split the integer and decimal parts in order to calculate seconds
+            # and nanos.
+            (integer_part, decimal_part) = duration_in_seconds.divmod 1
+            output.latency = Google::Protobuf::Duration.new(
+              seconds: integer_part,
+              nanos: (decimal_part * NANOS_IN_A_SECOND).round
+            )
+          else
+            # Logging API only takes a string with digits that ends with 's'.
+            output.latency = "#{duration_in_seconds.round(2)}s"
```

Comment:
You can send a hash with `seconds` and `nanos` here as well, just like we do for the timestamp above.